### ISSUE TITLE
Not able to write to Java files with vim

### ISSUE THREAD

**Issue by KaveenE** [nus-cs2030/2021-s2#8]
### Summary:
Can't seem to write to files with .java extension with vim if I didn't create them

**Description:**
I can read the files perfectly. I can read and write files that I created. I just can't seem to write to them if I personally didn't create them.
Eg: I download some code from Internet; any attempts to modify it results in an error

**Temporary fix**(I found the temporary fixes by googling so I don't know what I'm doing):
1) I used the command  `sudo vim` instead of `vim `to write. 
2) I tried to change mode of .java files to be "writeable and readable" with this command `chmod u+rw *.java`

**Does anyone know the proper solution?**

**Comment by DuelToDeath** [comment]
What is the error you get when you write to the files?

**Comment by KaveenE** [comment]
It says "Permission denied". 

Not sure if this helps but when I do `ls -l`,  the leftmost side of the output is` ----------` for the file I download. When I do `chmod u+rw *.java`, that changes to ` -rw-------`.

The files I create myself have ` -rw-r--r--`

**Comment by DuelToDeath** [comment]
Yeah seems like you don't have the permissions to write to the file. You may want to try chmod or chown

**Comment by KaveenE** [comment]
Sadly that's what I'm doing, does this happen to you as well as is it just a local problem?

**Comment by clementyee303** [comment]
Well if you really want to write to all future java files copied from outside sources without giving permissions individually you can try this. 
sudo chmod -R 777 /path/to/file/or/directory

not really recommended if you're concerned about data security :P

**Comment by KaveenE** [comment]
Oh, that's actually quite handy, thanks
